api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineWithPermissionsTest.java

```diff
@@ -71,80 +78,159 @@
+        expectedMap.put(pipeline2.getId(), permissionsForPipeline2);
 
-        grantPermissionsToPipeline(AclPermission.WRITE.getMask(), testUserSid2.getSid(), pipeline1.getId());
-        permissionsForPipeline1.put(TEST_OWNER2.toUpperCase(), 4);
         assertPipelineWithPermissions(expectedMap);
+    }
+
+    @Test
+    public void shouldReturnPermissionsWhenGrantedToPipelineCase3() {
+        final List<Pair<AbstractSecuredEntity, List<AbstractGrantPermission>>> pipelinesAndFolders = Arrays.asList(
+                new ImmutablePair<>(pipeline1, Arrays.asList(
+                        new UserPermission(TEST_USER_1, AclPermission.WRITE.getMask()),
+                        new UserPermission(TEST_USER_2, AclPermission.WRITE.getMask()))),
+                new ImmutablePair<>(pipeline2, Collections.singletonList(new UserPermission(TEST_USER_1, AclPermission.READ.getMask()))),
+                new ImmutablePair<>(folder1, Arrays.asList(
+                        new UserPermission(TEST_USER_1, AclPermission.READ.getMask()),
+                        new UserPermission(TEST_USER_2, AclPermission.READ.getMask()))),
+                new ImmutablePair<>(folder2, Collections.singletonList(new UserPermission(TEST_USER_1, AclPermission.READ.getMask()))));
+
+        mockPipelineAndFolders(pipelinesAndFolders);
+
+        permissionsForPipeline1.put(TEST_USER_1.toUpperCase(), 5);
+        permissionsForPipeline1.put(TEST_USER_2.toUpperCase(), 4);
+        permissionsForPipeline2.put(TEST_USER_2.toUpperCase(), 1);
+        permissionsForPipeline2.put(TEST_USER_1.toUpperCase(), 1);
+
+        expectedMap.put(pipeline1.getId(), permissionsForPipeline1);
+        expectedMap.put(pipeline2.getId(), permissionsForPipeline2);
 
-        grantPermissionToFolder(AclPermission.READ.getMask(), testUserSid2.getSid(), folder1.getId());
-        permissionsForPipeline1.put(TEST_OWNER2.toUpperCase(), 5);
-        permissionsForPipeline2.put(TEST_OWNER2.toUpperCase(), 1);
         assertPipelineWithPermissions(expectedMap);
+    }
+
+    @Test
+    public void shouldReturnPermissionsWhenGrantedToPipelineCase4() {
+        final List<Pair<AbstractSecuredEntity, List<AbstractGrantPermission>>> pipelinesAndFolders = Arrays.asList(
+                new ImmutablePair<>(pipeline1, Arrays.asList(
+                        new UserPermission(TEST_USER_1, AclPermission.WRITE.getMask()),
+                        new UserPermission(TEST_USER_2, AclPermission.WRITE.getMask()))),
+                new ImmutablePair<>(pipeline2, Collections.singletonList(new UserPermission(TEST_USER_1, AclPermission.READ.getMask()))),
+                new ImmutablePair<>(folder1, Arrays.asList(
+                        new UserPermission(TEST_USER_1, AclPermission.READ.getMask()),
+                        new UserPermission(TEST_USER_2, AclPermission.READ.getMask()),
+                        new UserPermission(TEST_USER_2, AclPermission.NO_READ.getMask()))),
```

Comment:
Why do you set both `READ` and `NO_READ` permissions? According to [the original test](https://github.com/epam/cloud-pipeline/blob/develop/api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineWithPermissionsTest.java#L138) you should grant `NO_READ` permission for the first folder to second user and then check that resulting mask equals to `2`.

Could you please double check that you are testing exact the same cases as the original test does in all test methods.